python/build_definitions/clockbound.py

```diff
@@ -0,0 +1,81 @@
+#
+# Copyright (c) YugabyteDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+# in compliance with the License. You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License
+# is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+# or implied. See the License for the specific language governing permissions and limitations
+# under the License.
+#
+
+from yugabyte_db_thirdparty.build_definition_helpers import *
+
+import os
+import shutil
+
+
+class ClockBoundDependency(Dependency):
+    """
+    aws/clock-bound is a convenience library to retrieve a strict ambiguity
+    time window within which the current reference clock's timestamp
+    is guaranteed to be present.
+
+    YB links target/release/libclockbound.so shared lib that provides the above
+    implementation.
+
+    YB includes clock-bound-ffi/include/clockbound.h header file that provides
+    the corresponding C interface.
+    """
+    def __init__(self) -> None:
+        super(ClockBoundDependency, self).__init__(
+            name='clockbound',
+            version='main',
+            url_pattern='https://github.com/aws/clock-bound/archive/{0}.zip',
+            build_group=BuildGroup.COMMON)
+
+    def build(self, builder: BuilderInterface) -> None:
+        with PushDir(builder.fs_layout.get_source_path(self)):
+            self.build_clockbound_ffi(builder)
+        self.lib_install_clockbound_ffi(builder)
+        self.hdr_install_clockbound_ffi(builder)
+
+    def build_clockbound_ffi(self, builder: BuilderInterface) -> None:
+        """ cargo build --release """
+        if not builder.prepare_for_build_tool_invocation(self):
+            return
+        log_prefix = builder.log_prefix(self)
+        log("Building dependency %s using cargo", self.name)
+
+        # Find the cargo binary in $HOME/.cargo/bin
+        cargo_bin = os.path.expanduser("~/.cargo/bin/cargo")
```

Comment:
This is what the official rust installation doc says

> In the Rust development environment, all tools are installed to the ~/.cargo/bin directory, and this is where you will find the Rust toolchain, including rustc, cargo, and rustup.

Source: https://www.rust-lang.org/tools/install

Unfortunately, Rust installation does not add its binaries to the PATH variable (probably because it does not require sudo permissions).

> This needs to work in various dev environments.
1. The tests run successfully which means that rust is installed in ~/.cargo/bin in all these environments.
2. Rust official doc says that the binaries are installed in ~/.cargo/bin/